iree/compiler/Conversion/LinalgToSPIRV/LinalgTileAndFusePass.cpp

```diff
@@ -408,28 +408,55 @@ static void populateTilingToInvocationPatterns(
                                  OpBuilder &builder, Location loc,
                                  ArrayRef<Range> parallelLoopRanges) {
     Type indexType = builder.getIndexType();
-    SmallVector<linalg::ProcInfo, 2> procInfo(2);
-    procInfo[1] = {builder.create<gpu::ThreadIdOp>(loc, indexType,
-                                                   builder.getStringAttr("x")),
-                   builder.create<ConstantIndexOp>(
-                       loc, launchConfig.getWorkgroupSize()[0])};
-    procInfo[0] = {builder.create<gpu::ThreadIdOp>(loc, indexType,
-                                                   builder.getStringAttr("y")),
-                   builder.create<ConstantIndexOp>(
-                       loc, launchConfig.getWorkgroupSize()[1])};
+    size_t numParallelLoop = parallelLoopRanges.size();
+    assert(numParallelLoop <= 3);
+    SmallVector<linalg::ProcInfo, 3> procInfo(numParallelLoop);
+    procInfo[numParallelLoop - 1] = {
```

Comment:
Similar comment here as #3990 . We should be able to reuse `getGPUProcessorIdsAndCounts`? Given that we are distributing with numprocs >= numiters, the size should not be used at all. In any case, we should always rely on using `gpu::BlockDimOp`.

Reply:
Good point, changed it.